src/index/base.cpp

```diff
@@ -168,7 +168,7 @@ void BaseIndex::ThreadSync()
             }
 
             if (last_locator_write_time + SYNC_LOCATOR_WRITE_INTERVAL < current_time) {
-                SetBestBlockIndex(pindex);
+                SetBestBlockIndex(pindex->pprev);
```

Comment:
In commit "index: Don't commit a best block before indexing it during sync" (9d4509b64ee333ef4587c77fb0d52e8e720a28c4)

I think commit description saying previous code leaves the "index database temporarily in an inconsistent state" is a little misleading, because the bug could permanently corrupt the index if node is killed before the block is committed. I'd drop the word "temporarily" and maybe mention that this could lead to corruption if the sync is interrupted at a bad time.

Also could add: 

```
Co-authored-by: Luke Dashjr <luke-jr+git@utopios.org>
```

since he spotted the simpler fix

Reply:
Done both. I meant "temporarily" as in "until the committed best block is indexed" and made that more clear in the commit description.